test/stubs/StubRuleViewer.ts

```diff
@@ -0,0 +1,19 @@
+import {Rule} from '@salesforce/code-analyzer-core';
+import {RuleViewer} from '../../src/lib/viewers/RuleViewer';
+
+
+export class StubRuleViewer implements RuleViewer {
+	private viewedRules: Rule[] = [];
+
+	public view(rules: Rule[]): void {
+		this.viewedRules.push(...rules);
+	}
+
+	public expectViewedRules(ruleNames: string[]): void {
```

Comment:
It is a little odd to have assertions inside of a stub. This is typically something you put in the test.

Stubs typically only are for the exercise phase of a 4 phased unit test (setup, exercise, verify, teardown).

Reply:
Makes sense. My reasoning was that I wanted to avoid leaking details of how the calls are stored up into the tests themselves, but it should still be pretty clean if I do it the other way.